changelog/392.doc.rst

```diff
@@ -0,0 +1,6 @@
+Split up documentation into multiple files.
+- Main ``disnake`` API reference is now under ``api/``.
+- ``disnake.ext.commands`` API reference is now under ``ext/commands/api/``.
+- ``disnake.ext.tasks`` is considered small enough to leave it as-is, and is not changed.
+- Legacy ``api.html`` and ``ext/commands/api.html`` pages are deprecated and should not be used anymore.
+- - They redirect previously valid URLs to the new object locations for now, but this can be changed later at any time without prior notice.
```

Comment:
Not entirely sure if these are worth mentioning here, the redirects exist for existing links and and people shouldn't try to use them.

Reply:
IMO it is.
Also notice
> this can be changed later at any time without prior notice.
Meaning we can make a pr removing redirects at any point without !breaking marker and changelog entry.
This is also would reduce potential "why api.html dont work!?!?" questions on discord.

Reply:
I don't see a reason to remove the redirects in the future, fragment names should always be unique - the `this can be changed later` warning isn't necessary.

Reply:
The biggest reason for removing redirects is the size they take when loading built page: docs data in compressed json takes ~0.5MB right now, and more things come, more it will take. This is a really big deal for slow networks, and it is actually even worse because this script is loaded *on every page*, not only api.html or ext/cmds/api.html

Reply:
Is it possible to only load that script on those pages? This somewhat loses some of the benefit these changes provide us.

Reply:
The most obvious (but not best) solution would be adding a few `if`s to jinja templates, but ill try to find something better

Reply:
Done in [`23a48d4`](https://github.com/DisnakeDev/disnake/pull/722/commits/23a48d4335d5c029358cf13bdb803cb604fd0eae)